library/manageiq_tag_assignment.py

```diff
@@ -0,0 +1,225 @@
+author: Daniel Korn (@dkorn)
+options:
+  miq_url:
+    description:
+      - the manageiq environment url
+    default: MIQ_URL env var if set. otherwise, it is required to pass it
+  miq_username:
+    description:
+      - manageiq username
+    default: MIQ_USERNAME env var if set. otherwise, it is required to pass it
+  miq_password:
+    description:
+      - manageiq password
+    default: MIQ_PASSWORD env var if set. otherwise, it is required to pass it
+  tags:
+    description:
+      - list of dictionaries representing tags, each includes the tag name and category name
+      - categories have a 'single_value': True/False. assigning a tag in a single_value: True category
+        clears other tags from that category.
+    required: true
+    default: null
+  resource:
+    description:
+      - the relevant resource type in manageiq
+    required: true
+    choices: ['provider', 'host', 'vm', 'blueprint', 'category', 'cluster', 'data store', 'group', 'resource pool', 'service', 'service template', 'template', 'tenant', 'user']
+    default: null
+  resource_name:
+    description:
+      - the relevant resource name in manageiq
+    required: true
+    default: null
+  state:
+    description:
+      - On present, it will assign the tag on the resource, if not already assigned
+      - On absent, it will unassign the tag from the resource
+    required: true
+    choices: ['present', 'absent']
+    default: null
+  miq_verify_ssl:
+    description:
+      - whether SSL certificates should be verified for HTTPS requests
+    required: false
+    default: True
+    choices: ['True', 'False']
+  ca_bundle_path:
+    description:
+      - the path to a CA_BUNDLE file or directory with certificates
+    required: false
+    default: null
+'''
+
+EXAMPLES = '''
+# Assign tag on a provider in ManageIQ
+  manageiq_tag_assignment:
+    tags:
+    - category: environment
+      name: prod
+    - category: owner
+      name: prod_ops
+    resource_name: 'OpenShift01'
+    resource: 'provider'
+    state: 'present'
+    miq_url: 'http://localhost:3000'
+    miq_username: 'admin'
+    miq_password: '******'
+'''
+
+
+class ManageIQ(object):
+    """ ManageIQ object to execute tag assignments in manageiq
+
+    url            - manageiq environment url
+    user           - the username in manageiq
+    password       - the user password in manageiq
+    miq_verify_ssl - whether SSL certificates should be verified for HTTPS requests
+    ca_bundle_path - the path to a CA_BUNDLE file or directory with certificates
+    """
+
+    manageiq_entities = {
+        'provider': 'providers', 'host': 'hosts', 'vm': 'vms',
+        'category': 'categories', 'cluster': 'clusters', 'data store': 'data_stores',
+        'group': 'groups', 'resource pool': 'resource_pools', 'service': 'services',
+        'service template': 'service_templates', 'template': 'templates',
+        'tenant': 'tenants', 'user': 'users', 'blueprint': 'blueprints'
+    }
```

Comment:
We already have these somewhere (policy assignment?).  Can you move it to a shared module?